waku/v2/node/message_store.nim

```diff
@@ -10,8 +10,6 @@ import
   ../waku_types,
   ./sqlite
 
-{.push raises: [Defect].}
```

Comment:
You probably do want this, if you're getting errors around unhandled exceptions, it's probably because there are unhandled exceptions and they will cause issues in the long run, specially if using `Result`.

Reply:
@dryajov this was mainly on the callback function DataProc, it didn't seem to ever really call an exception. We do still have this in the database section however.

Reply:
you need to annotate the `DataProc` alias specifically with `raises: [Defect]`  so that anyone that provides a callback is forced to handle their exceptions, else the compiler must make a conservative estimate

Reply:
see `kvstore.nim`